packages/roosterjs-editor-core/lib/corePlugins/CopyPastePlugin.ts

```diff
@@ -86,10 +92,20 @@ export default class CopyPastePlugin implements PluginWithState<CopyPastePluginS
                 html,
                 this.editor.getTrustedHTMLHandler()
             );
-            const newRange =
-                metadata?.type === SelectionRangeTypes.Normal
-                    ? createRange(tempDiv, metadata.start, metadata.end)
-                    : null;
+            let newRange;
+
+            if (selection.type === SelectionRangeTypes.TableSelection) {
+                const table = tempDiv.querySelector(`#${selection.table.id}`) as HTMLTableElement;
+                newRange = this.createTableRange(table, selection.coordinates);
+                if (isCut) {
+                    this.deleteTableContent(selection.table, selection.coordinates);
+                }
+            } else {
+                newRange =
+                    metadata?.type === SelectionRangeTypes.Normal
+                        ? createRange(tempDiv, metadata.start, metadata.end)
+                        : null;
+            }
 
             this.editor.triggerPluginEvent(PluginEventType.BeforeCutCopy, {
                 clonedRoot: tempDiv,
```

Comment:
From line 117, better use the event (returned from triggerPluginEvent from line 110) range property:
```ts
const event = this.editor.triggerPluginEvent(...);
if (event.range) {
 ...
}
```

Reply:
Because we allow plugin to replace the range object